### PR TITLE
Wi-Fi shell improvements

### PR DESCRIPTION
**Pull request by krish2718**

### CODE REVIEW

File: subsys/net/l2/wifi/wifi_shell.c [hunk 1]
```diff
@@ -1651,111 +1651,108 @@ SHELL_STATIC_SUBCMD_SET_CREATE(wifi_commands,
...
+		  ": 0:Disable, 1:Optional, 2:Required.\n",
 		  cmd_wifi_connect,
-		  2, 5),
-	SHELL_CMD_ARG(disconnect, NULL, "Disconnect from the Wi-Fi AP",
+		  2, 4),
+	SHELL_CMD_ARG(disconnect, NULL, "Disconnect from the Wi-Fi AP.\n",
 		  cmd_wifi_disconnect,
 		  1, 0),
-	SHELL_CMD_ARG(ps, NULL, "Configure Wi-F PS on/off, no arguments will dump config",
+	SHELL_CMD_ARG(ps, NULL, "Configure Wi-F power save state.\n"
```
Comment (sachinthegreen) [inline subsys/net/l2/wifi/wifi_shell.c:1660]:
```suggestion
	SHELL_CMD_ARG(ps, NULL, "Configure or display Wi-Fi power save state.\n"
```

**Review by sachinthegreen — CHANGES_REQUESTED** [review]

**Review by jukkar — DISMISSED** [review]

**Comment by krish2718** [comment]
FYI, for a single line change in help, had to pull in many of the `scan` re-work `fromtree` commits, as upmerge is still pending.

**Comment by jukkar** [comment]
> FYI, for a single line change in help, had to pull in many of the `scan` re-work `fromtree` commits, as upmerge is still pending.

Just wondering if you could delay the related ncs work until the upstream patch is merged, and only then do downstream commits etc.

**Review by jukkar — DISMISSED** [review]

**Comment by krish2718** [comment]
> > FYI, for a single line change in help, had to pull in many of the `scan` re-work `fromtree` commits, as upmerge is still pending.
> 
> Just wondering if you could delay the related ncs work until the upstream patch is merged, and only then do downstream commits etc.

The reason for doing this in tandem it to run the NCS CI before upstream PR is merged, this way we can catch any bugs easily. And once its merged, we can abandon the PR or merge it.

> FYI, for a single line change in help, had to pull in many of the `scan` re-work `fromtree` commits, as upmerge is still pending.

Just realized that this comment was supposed to be in downstream :-).

**Review by sachinthegreen — APPROVED** [review]

**Review by jukkar — DISMISSED** [review]

**Review by rlubos — DISMISSED** [review]
Looks good, but please check the twister results. Looks like some include is missing? At least  minimal libc declares `strncasecmp()` in `strings.h`.

**Review by jukkar — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**